### PR TITLE
Feature/fetch customer flow

### PR DESCRIPTION
**Pull request by andriibodryi**

### CODE REVIEW

File: src/components/Button/Button.test.tsx [hunk 1]
```diff
@@ -2,13 +2,28 @@
...
-    <Button onPress={() => ({})} text="Test Render" />,
-  );
-  expect(tree).toMatchSnapshot();
+jest.mock('react-native-reanimated', () =>
+  jest.requireActual('../../../node_modules/react-native-reanimated/mock'),
+);
+
+describe('Button component', () => {
+  let tree: RenderAPI;
+  const onPressMock = jest.fn();
```
Comment (stoyar) [inline src/components/Button/Button.test.tsx:14]: It is better to move the variable to the test where it is used

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
-  ViewStyle,
-  TextInputProps,
-} from 'react-native';
+import { TextInput, View, ViewStyle, TextInputProps } from 'react-native';
 import { Text } from '~/components/Text/Text';
 
 import { styles } from './styles';
 import { Colors } from '~/app-styles';
 import { useAnimatedLabel } from '~/components/Input/useAnimation';
+import { AnimatedText } from '~/hooks';
```
Comment (stoyar) [inline src/components/Input/Input.tsx:8]: Please, move it to the components

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -1,16 +1,11 @@
...
-  Animated,
-  ViewStyle,
-  TextInputProps,
-} from 'react-native';
+import { TextInput, View, ViewStyle, TextInputProps } from 'react-native';
 import { Text } from '~/components/Text/Text';
 
 import { styles } from './styles';
 import { Colors } from '~/app-styles';
 import { useAnimatedLabel } from '~/components/Input/useAnimation';
```
Comment (stoyar) [inline src/components/Input/Input.tsx:7]: Please, move it to the hooks

**Review by Mykhailo-Kaduk — APPROVED** [review]

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 4]
```diff
@@ -47,45 +45,55 @@ export const BottomSheetModal: React.FC<Props> = ({
...
     sheetRef.current?.snapTo(0);
   }, []);
 
-  const toggleBottomSheet = useCallback(() => {
-    requestAnimationFrame(() => {
-      setBottomSheetState((state) => !state);
-    });
-  }, []);
-
   const renderContent = () => (
```
Comment (stoyar) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:60]: Is it better to declare auxiliary functions outside of component declaration?

File: src/components/BottomSheetModal/BottomSheetModal.tsx [hunk 5]
```diff
@@ -47,45 +45,55 @@ export const BottomSheetModal: React.FC<Props> = ({
...
     <View style={styles.content}>
       {children || <Text>Swipe down to close</Text>}
     </View>
   );
 
-  const renderHeader = useCallback(() => <View style={styles.header} />, []);
+  const renderHeader = () => <View style={styles.header} />;
```
Comment (stoyar) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:66]: The same

File: src/components/BottomSheetModal/BottomSheetModal.tsx (hunk 4 above)
Comment (stoyar) [inline src/components/BottomSheetModal/BottomSheetModal.tsx:60]: Is it better to declare auxiliary functions outside of component declaration?

File: src/components/Loader/Loader.test.tsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable import/no-extraneous-dependencies */
+
+import React from 'react';
+import { render, RenderAPI } from '@testing-library/react-native';
+import { Loader } from './Loader';
+
+describe('Loader component', () => {
+  let tree: RenderAPI;
+  beforeEach(() => {
```
Comment (stoyar) [inline src/components/Loader/Loader.test.tsx:9]: You don't need beforeEach with single test case.

File: src/localization/english.ts [hunk 7]
```diff
@@ -1,15 +1,24 @@
...
   'en-US': {
     screenTitles: {
-      DASHBOARD_TITLE: 'DASHBOARD',
+      CUSTOMER_SCANNER_TITLE: 'Customer scanner',
       USER_TITLE: 'USER',
     },
     general: {
       LOADING: 'Loading...',
-      TO_DASHBOARD: 'Go to Dashboard',
+      TO_DASHBOARD: 'Go to CustomerScanner',
```
Comment (stoyar) [inline src/localization/english.ts:9]: Please rename this constant

File: src/localization/english.ts [hunk 8]
```diff
@@ -1,15 +1,24 @@
...
       TO_USER: 'Go to User Details',
       EMAIL: 'Email',
       PASSWORD: 'Password',
+      SAVE: 'Save',
```
Comment (stoyar) [inline src/localization/english.ts:13]: I think, create 'buttons' section for buttons' labels - good idea. Please take a look on the latest changes in the Localization component

File: src/localization/franch.ts [hunk 9]
```diff
@@ -1,7 +1,7 @@
-export const frenchStrings: Record<string, unknown> = {
+export const frenchStrings = {
```
Comment (stoyar) [inline src/localization/franch.ts:1]: As we talk about, french strings is not needed now

File: src/screens/CustomerScanner/components/CustomerInfoCard/CustomerInfoCard.tsx [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+          onPress={choseAnotherCustomer}
+          primary
+          medium
+          text={customerScanner.CHOSE_ANOTHER_CUSTOMER}
+        />
+      </AnimatedView>
+
+      <AnimatedView style={[styles.customerEdit, editingStyles]}>
+        <Text title gray>
+          Edit Customer Info
```
Comment (stoyar) [inline src/screens/CustomerScanner/components/CustomerInfoCard/CustomerInfoCard.tsx:88]: Please, use localization

File: src/screens/CustomerScanner/store/actionTypes.ts [hunk 11]
```diff
@@ -0,0 +1,8 @@
+export const RESET_CUSTOMER = '@customer/RESET_CUSTOMER';
```
Comment (stoyar) [inline src/screens/CustomerScanner/store/actionTypes.ts:1]: Why didn't you use enum to unite all types into one entity?

File: src/components/BottomSheetModal/BottomSheetModal.tsx (hunk 4 above)
Comment (andriibodryi) [reply src/components/BottomSheetModal/BottomSheetModal.tsx:60]: image
be more attentive, the function uses children from the props of the BottomSheetModal component. Do you propose to create a component and pass as prop the children one level below?

File: src/screens/CustomerScanner/store/actionTypes.ts (hunk 11 above)
Comment (andriibodryi) [reply src/screens/CustomerScanner/store/actionTypes.ts:1]: there is no point in using enum. const is enough

File: src/localization/english.ts (hunk 8 above)
Comment (andriibodryi) [reply src/localization/english.ts:13]: maybe
If you you want, you may create new pr for this feature

File: src/components/BottomSheetModal/BottomSheetModal.tsx (hunk 4 above)
Comment (stoyar) [reply src/components/BottomSheetModal/BottomSheetModal.tsx]: I meant, to declare in this file but outside of function declaration. It increases readability. Up to you)

**Review by stoyar — APPROVED** [review]

**Pull request merged**